### PR TITLE
Feat: add new works banner to my collection screen

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **Feature**

Builds off the initial work by @olerichter00 here: https://github.com/artsy/eigen/pull/5730

This PR resolves [CX-1868]

### Description

https://user-images.githubusercontent.com/18648835/140505719-d2a57d26-7613-427c-8c9d-00afe8fbcec1.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Added New works banner to my collection screen - ole, kizito

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1868]: https://artsyproduct.atlassian.net/browse/CX-1868?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  We are trying to migrate away from `react-test-renderer` towards `@testing-library/react-native`, but found Test-Renderer imports in the following new unit test files:

- `src/palette/elements/Banner/Banner.tests.tsx`

See [`Pill.tests.tsx`](https://github.com/artsy/eigen/blob/2f32d462bb3b4ce358c8a14e3ed09b42523de8bd/src/palette/elements/Pill/__tests__/Pill-tests.tsx) as an example, or [the docs](https://callstack.github.io/react-native-testing-library/docs/api-queries).
  
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6b35fccd27300a792195c3bad08d79e934013c90
</p>

**Review by brainbicycle — APPROVED** [review]
🎉 nice, thanks for making reusable

**Review by pvinis — APPROVED** [review]
very nice! I would like the asyncstorage thing to just be a boolean in the global store, so that persistence is handled there automatically, but I won't block this PR.
If we keep it separate as here, I'd like to see a ticket on jira to move this at some point, and maybe we could look for other things that are similar, like app launch count etc.

File: src/lib/Scenes/MyCollection/MyCollection.tsx [hunk 1]
```diff
@@ -41,6 +42,13 @@ export function unsafe_getEnableMyCollection() {
   return unsafe_getFeatureFlag(featureFlagKey)
 }
 
+export const HAS_SEEN_MY_COLLECTION_NEW_WORKS_BANNER = "HAS_SEEN_MY_COLLECTION_NEW_WORKS_BANNER"
+
+const hasBeenShownBanner = async () => {
+  const hasSeen = await AsyncStorage.getItem(HAS_SEEN_MY_COLLECTION_NEW_WORKS_BANNER)
```
[the remark this answers is not in the document]
Comment (lordkiz) [reply src/lib/Scenes/MyCollection/MyCollection.tsx:48]: Thanks @pvinis I agree! I couldn't find an existing model well suited for it. Would you suggest creating a stats model to hold these kind of data? The closest existing model I might want to put this in is EmissionState. Even then, it feels like it doesn't belong there.

[the remark this answers is not in the document]
Comment (lordkiz) [reply src/lib/Scenes/MyCollection/MyCollection.tsx:48]: @pvinis will be done here: https://artsyproduct.atlassian.net/browse/CX-2151

**Comment by brainbicycle** [comment]
alright we have a ticket for the async storage concern: https://artsyproduct.atlassian.net/browse/CX-2151 which I agree on, going to go ahead and merge so we can have a build to QA today!

**Pull request merged**